dali/kernels/signal/fft/fft_postprocess.cuh

```diff
@@ -0,0 +1,385 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_SIGNAL_FFT_FFT_POSTPROCESS_CUH_
+#define DALI_KERNELS_SIGNAL_FFT_FFT_POSTPROCESS_CUH_
+
+#include <cuda_runtime.h>
+#include "dali/core/geom/vec.h"
+#include "dali/kernels/kernel.h"
+#include "dali/kernels/signal/fft/fft_common.h"
+
+namespace dali {
+namespace kernels {
+namespace signal {
+
+namespace fft_postprocess {
+
+using namespace fft;  // NOLINT
+
+template <typename Out, typename In>
+struct SampleDesc {
+  Out *out;
+  const In *in;
+  ptrdiff_t out_stride;
+  ptrdiff_t in_stride;
+};
+
+struct BlockDesc {
+  int sample_idx;
+  ivec2 start, end;
+};
+
+constexpr int kBlock = 32;
+
+struct norm2square {
+  DALI_HOST_DEV DALI_FORCEINLINE
+  auto operator()(float2 c) const {
+    return c.x * c.x + c.y * c.y;
+  }
+
+  DALI_FORCEINLINE
+  auto operator()(complexf c) const {
+    return (*this)(float2{c.real(), c.imag()});
+  }
+};
+
+struct norm2 {
+  DALI_HOST_DEV DALI_FORCEINLINE
+  auto operator()(float2 c) const {
+#ifdef __CUDA_ARCH__
+    return sqrtf(c.x * c.x + c.y * c.y);
+#else
+    return std::sqrt(c.x * c.x + c.y * c.y);
+#endif
+  }
+
+  DALI_FORCEINLINE
+  auto operator()(complexf c) const {
+    return (*this)(float2{c.real(), c.imag()});
+  }
+};
+
+struct power_dB {
```

Comment:
Isn't that common for the CPU part as well?

Reply:
I don't think there's an end-to-end utility that would take complex numbers and produce decibels - much less one usable on the GPU, where std::complex is, unfortunately, not supported.

Reply:
Also, CPU has the reference level - it probably should be here, too, but for that we'd need min/max reduction kernels for GPU (preferably batched) which is far beyond the scope of this PR.